internal/fs/handle/dir_handle_test.go

```diff
@@ -0,0 +1,138 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/contentcache"
+	"github.com/googlecloudplatform/gcsfuse/internal/fs/inode"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	"github.com/jacobsa/gcloud/gcs/gcsutil"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+)
+
+func TestDirHandle(t *testing.T) { RunTests(t) }
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type DirHandleTest struct {
+	ctx    context.Context
+	bucket gcsx.SyncerBucket
+	clock  timeutil.SimulatedClock
+
+	dh *DirHandle
+}
+
+var _ SetUpInterface = &DirHandleTest{}
+var _ TearDownInterface = &DirHandleTest{}
+
+func init() { RegisterTestSuite(&DirHandleTest{}) }
+
+func (t *DirHandleTest) SetUp(ti *TestInfo) {
+	t.ctx = ti.Ctx
+	t.bucket = gcsx.NewSyncerBucket(
+		1, ".gcsfuse_tmp/", gcsfake.NewFakeBucket(&t.clock, "some_bucket"))
+	t.clock.SetTime(time.Date(2022, 8, 15, 22, 56, 0, 0, time.Local))
+	t.resetDirHandle()
+}
+
+func (t *DirHandleTest) TearDown() {}
+
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+func (t *DirHandleTest) resetDirHandle() {
+	dirInode := inode.NewDirInode(
+		17,
+		inode.NewDirName(inode.NewRootName(""), "mnt"),
+		fuseops.InodeAttributes{
+			Uid:  123,
+			Gid:  456,
+			Mode: 0712,
+		},
+		false, // implicitDirs
+		false, // enableNonExistentTypeCache
+		0,     //typeCacheTTL
+		&t.bucket,
+		&t.clock,
+		&t.clock)
+
+	t.dh = NewDirHandle(
+		dirInode,
+		true,
+	)
+}
+
+func (t *DirHandleTest) createLocalFileInode(name string, id fuseops.InodeID) (in inode.Inode) {
+	in = inode.NewFileInode(
+		id,
+		inode.NewFileName(t.dh.in.Name(), name),
+		nil,
+		fuseops.InodeAttributes{
+			Uid:  123,
+			Gid:  456,
+			Mode: 0712,
+		},
+		&t.bucket,
+		false, // localFileCache
+		contentcache.New("", &t.clock),
+		&t.clock,
+		true) // localFile
+	return
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *DirHandleTest) EnsureEntriesWithLocalAndGCSFiles() {
```

Comment:
Test cases when 
1. local fileinode and GCS object has same name
2. local fileinode and GCS directory has same name